obhod_lune/obhod.py

```diff
@@ -0,0 +1,533 @@
+                    break
+
+                if sat_y < -0.25 or sat_y > 0.25:  # Sling not strong enough - didn't get turned around properly
+                    too_low = True
+                    break
+
+                if sat_x < mu_line:  # Passed Moon 2nd time - above
+                    if sat_y > 0:
+                        iii_passed = True
+                        """
+                        if abs(sat_y + y_in) < 1e-7:
+                            iii_passed = True
+
+                        if sat_y < y_in : # Got pulled in too tightly
+                            too_high = True
+                            break
+
+                        if sat_y > y_in : # Got pulled in too weakly
+                            too_low = True
+                            break
+                        """
+
+            elif not iv_passed:
+                if sat_x > mu_line:  # Passed Moon again
+                    if sat_y < 0 : # Below
+                        iv_passed = True
+                        print('Found something?')
+                        break
+                        """
+                        if abs(sat_y - y_in) < 1e-7:
+                            iv_passed = True
+                            print('Found something?')
+                            break
+
+                        if sat_y < y_in : # Got pulled in too tightly
+                            too_high = True
+                            break
+
+                        if sat_y > y_in : # Got pulled in too weakly
+                            too_low = True
+                            break
+                        """
+                    else : # otherwise pull not strong enough
+                        too_low = True
+                        break
+
+                if sat_x < -mu:  # Passed towards Earth
+                    too_low = True
+                    break
+
+                if sat_y > 0.25:  # Passed above Moon and overshot above the domain limit above - sling not strong enough
+                    too_low = True
+                    break
+
+            else:
+                print('Found something?')
+                break
+
+        if too_low:
+            low_dy = mid_dy
+        elif too_high:
+            high_dy = mid_dy
+        else:
+            print('What do now?')
+            print("Current best guess {}".format(mid_dy))
+            print("I. {}, II. {}, III. {}, IV. {}".format(i_passed, ii_passed, iii_passed, iv_passed))
+            break
+
+        print("Current best guess {}".format(mid_dy))
+        print(
+            "I. {}, II. {}, III. {}, IV. {}".format(i_passed, ii_passed, iii_passed, iv_passed))
+
+    return mid_dy
+
+# Draw the simulation with the specified start parameters
+def draw_simulation(h, mu, mu_line, start_x, start_dy, n_steps) :
+    x = [start_x]
+    y = [0]
+
+    satellite = Satellite(start_x=start_x, start_y=0.0, start_dx=0.0, start_dy=start_dy, mu=mu, mu_line=mu_line)
+
+    for i in range(n_steps):
```

Comment:
I would extract calculating steps into a separate method. This way you can easily add tests, error estimations etc.